### PR TITLE
Implemented AWS Lambda & API Gateway & RDS MySql Integration + Kafka Streaming Data Analysis

### PR DESCRIPTION
**Pull request by quanganhhoang**
Merging to master so we have something we all can present on. Let me know if you can replicate my setup and/or if you have suggestions on documentation.

### CODE REVIEW

File: tutorial/aws-serverless-demo/SkierApiFunction/src/main/java/handlers/ResortService.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import model.Resort;
+import model.ResortList;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import sql.HikariDS;
+
+public class ResortService implements RequestHandler<Object, GatewayResponse> {
+  private static final Logger logger = LogManager.getLogger(ResortService.class);
```
Comment (irgregg) [inline tutorial/aws-serverless-demo/SkierApiFunction/src/main/java/handlers/ResortService.java]: Context has a logger in it, so no need to instantiate one here.

File: tutorial/aws-serverless-demo/SkierApiFunction/src/main/java/handlers/SkierService.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import model.PostBody;
+import model.SkierPostPathParam;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import sql.HikariDS;
+
+public class SkierService implements RequestHandler<GatewayRequest, GatewayResponse> {
+  private static final Logger logger = LogManager.getLogger(ResortService.class);
```
Comment (irgregg) [inline tutorial/aws-serverless-demo/SkierApiFunction/src/main/java/handlers/SkierService.java]: Same as previous

File: tutorial/aws-serverless-demo/SkierApiFunction/src/main/java/sql/HikariDS.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
+package sql;
+
+import com.zaxxer.hikari.HikariDataSource;
+import java.sql.Connection;
+import java.sql.SQLException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class HikariDS {
```
Comment (irgregg) [inline tutorial/aws-serverless-demo/SkierApiFunction/src/main/java/sql/HikariDS.java]: This class still strikes me as really wasteful given that only one request will be running on a single lambda instance at any given time.

File: tutorial/aws-serverless-demo/SkierApiFunction/src/main/java/handlers/ResortService.java (hunk 1 above)
Comment (quanganhhoang) [reply tutorial/aws-serverless-demo/SkierApiFunction/src/main/java/handlers/ResortService.java]: Yep I was going back and forth between using Lambda Logger or Log4j. AWS Lambda actually recommends log4j2 for a custom log appender! 

https://docs.aws.amazon.com/lambda/latest/dg/java-logging.html

File: tutorial/aws-serverless-demo/SkierApiFunction/src/main/java/sql/HikariDS.java (hunk 3 above)
Comment (quanganhhoang) [reply tutorial/aws-serverless-demo/SkierApiFunction/src/main/java/sql/HikariDS.java]: I agree! Your approach makes more sense. Let's mention this in our presentation.

**Pull request merged**